drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -849,12 +864,13 @@ static void usbd_event_transfer_ctrl(nrfx_usbd_evt_t const *const p_event)
 	if (NRF_USBD_EPIN_CHECK(p_event->data.eptransfer.ep)) {
 		switch (p_event->data.eptransfer.status) {
 		case NRFX_USBD_EP_OK: {
-			struct usbd_ep_event *ev = usbd_evt_alloc();
-
-			ev->ep = ep_ctx;
-			ev->evt = EP_EVT_WRITE_COMPLETE;
+			struct usbd_event *ev = usbd_evt_alloc();
```

Comment:
allocation can fail (also in other cases)

Reply:
Changes applied.
Queue size is now configurable. If allocation fails then the drivers is reinitialized and queue is flushed with an error message in log. Device can still be re-enumerated - a "soft" form of a failure instead of device reset as it was before.